R/timeseries_plot.R

```diff
@@ -0,0 +1,100 @@
+# multiple stations (month)
+niger_data <- readRDS("C:/Users/lzc1n17/OneDrive - University of Southampton/PhD/IDEMS/RInstatClimatic/Niger4_day_month_year.RDS")
+niger_data <- niger_data$get_data_frame()
+niger_data_1 <- niger_data$Data
+niger_monthly_data <- niger_data$Data_by_year_month
+niger_yearly_data <- niger_data$Data_by_year
+niger_monthly_data$station = ifelse(niger_monthly_data$month_abbr %in% c("Jan", "Feb", "Mar"), "A", "B")
+
+
+# currently: ggplot, ggmiscp, viridis
+  
+
+# facets = none ----------------------
+
+ts_line_plot <- function(data, time = year, element, station = NULL, facets = "none",
+                         add_points = FALSE, add_line_of_best_fit = FALSE, add_peaks = FALSE, add_valleys = FALSE,
+                         se = TRUE, add_path = FALSE, add_step = FALSE){
+  
+  # this is only applied when there is more than one element
+  data_longer <- data %>% tidyr::pivot_longer(cols = c({{ element }}), names_to = "Element")
+  
+  if (facets == "none"){
+    
+    if (length(data %>% dplyr::select({{ element }})) == 1){ # what is the correct way to check the length!
+      base_plot <- ggplot(data, mapping = aes(x = {{ time }}, y = {{ element }}, colour = {{ station }})) +
+        geom_line()
+    } else {
+      if (missing(station)){
+        base_plot <- ggplot(data_longer, mapping = aes(x = {{ time }}, y = value, colour = Element))
+      } else {
+        data_longer <- data_longer %>%
+          mutate(station_element = paste(station, Element, sep = "_"))
+        base_plot <- ggplot(data_longer, mapping = aes(x = {{ time }}, y = value, colour = station_element))
+      }
+    }
+  }
+  if (facets == "elements"){
+    base_plot <- ggplot(data_longer, mapping = aes(x = {{ time }}, y = value, colour = {{ station }})) +
+      facet_grid(cols = vars(Element))
+  } else if (facets == "stations"){
+  if (length(data %>% dplyr::select({{ element }})) == 1){
+    base_plot <- ggplot(data, mapping = aes(x = {{ time }}, y = {{ element }}))
+  } else {
+    base_plot <- ggplot(data_longer, mapping = aes(x = {{ time }}, y = value, colour = Element))
+  }
+  
+  base_plot <- base_plot +
+    facet_grid(cols = vars({{ station }}))
+} else if (facets == "both"){
+  base_plot <- ggplot(data_longer, mapping = aes(x = {{ time }}, y = value)) +
+    facet_grid({{ station }} ~ Element)
+}
+  
+  base_plot <- base_plot + geom_line()
+  
+  # color by viridis
+  base_plot <- base_plot + viridis::scale_colour_viridis(discrete = TRUE, option = "C") # colour blind friendly
+  
+  if (add_points){
+    base_plot <- base_plot + geom_point()
+  }
+  
+  if (add_line_of_best_fit){
+    base_plot <- base_plot + geom_smooth(method = "lm", se = se)
+  }
+  
+  if (add_peaks){
+    base_plot <- base_plot + ggpmisc::stat_peaks(geom = "text", colour = "red")
+  }
+  
+  if (add_valleys){
+    base_plot <- base_plot + ggpmisc::stat_valleys(geom = "text", colour = "blue")
```

Comment:
I wonder how important these are and if we might rid of these because it means including another package dependency just for this?

Reply:
Agreed. The user can always add them on if they wish to "manually"